### ISSUE TITLE
Runners and `wasmer_wasix::wapm` don't take the `[fs]` table into account

### ISSUE THREAD

**Issue by Michael-F-Bryan** [wasmerio/wasmer#3744]
### Describe the bug

The current runner implementation (used by `wasmer run` and `wasmer run-unstable`) don't respect the directory mappings in the `wasmer.toml` file's `[fs]` table.

### Steps to reproduce

The easiest way to reproduce this is by creating a package that will serve static files, where the original location on the publishing machine's filesystem doesn't match the directory the WASI executable will see.

```toml
# wasmer.toml

[package]
name = "Michael-F-Bryan/wasmer-io"
version = "0.2.0"
description = "The wasmer.io website"

[dependencies]
"sharrattj/static-web-server" = "1"

[fs]
public = "out"
```

### Expected Solution

There are two ways we could resolve this.

The first one is by making sure consumers of a `*.webc` file respect the `[fs]` mapping. At the moment our runners use the naive `WebcFileSystem::init_all()` method to set up the volumes ([here](https://github.com/wasmerio/wasmer/blob/35d9cf623639c97d3b6a001a047441d9b2c08321/lib/wasi/src/runners/container.rs#L112-L126)), and that just implements a poor man's unionfs with all the volumes.

Alternatively, the `wapm-targz-to-pirita` could shuffle directories around as part of the general tarball-to-webc conversion.

**Comment by Michael-F-Bryan** [comment]
We should also check that people can't do something nefarious like `[fs] public = "/etc/shadow"`.

**Comment by Michael-F-Bryan** [comment]
I haven't implemented it in #3852, but I've left some placeholders for the solution.

This is where the resolver should figure out how volumes should be mapped:

https://github.com/wasmerio/wasmer/blob/8207c5558857de2a541a92a3a3a9a8081c66a4b1/lib/wasi/src/runtime/resolver/resolve.rs#L211-L218

And this is where we take WEBC containers and do the actual mapping:

https://github.com/wasmerio/wasmer/blob/8207c5558857de2a541a92a3a3a9a8081c66a4b1/lib/wasi/src/runtime/package_loader/load_package_tree.rs#L232-L241
